src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java

```diff
@@ -0,0 +1,1202 @@
+                os.append(achIncomeNote.getNoteText());
+                os.append("\n");
+            }
+            os.close();
+
+            Note note = documentService.createNoteFromDocument(document, "");
+            String attachmentType = null;
+            BufferedInputStream fileStream = new BufferedInputStream(new FileInputStream(attachmentFile));
+            File file = new File(attachmentFile);
+            Attachment attachment = attachmentService.createAttachment(document.getDocumentHeader(), file.getName(), "text", (int) file.length(), fileStream, attachmentType);
+            note.setAttachment(attachment);
+            attachment.setNote(note);
+            noteService.save(note);
+        } catch (Exception e) {
+            LOG.error("Error while adding notes to advance deposit documents: " + e.getMessage(), e);
+            throw new RuntimeException("Error while adding notes to the document " + document.getDocumentNumber());
+        } finally {
+            if (os != null) {
+                try {
+                    os.close();
+                } catch (IOException ie) {
+                    // not really much to do at this point
+                }
+            }
+        }
+    }
+
+    protected void createSourceAccountingLine(AchIncomeTransaction transaction, AdvanceDepositDocument advanceDepositDocument) {
+        String chart = parameterService.getParameterValueAsString(GenerateAdvanceDepositDocumentsStep.class, CuFPParameterConstants.AdvanceDepositDocument.CHART);
+        String objectCode = parameterService.getParameterValueAsString(GenerateAdvanceDepositDocumentsStep.class, CuFPParameterConstants.AdvanceDepositDocument.OBJECT_CODE);
+        String account = parameterService.getParameterValueAsString(GenerateAdvanceDepositDocumentsStep.class, CuFPParameterConstants.AdvanceDepositDocument.ACCOUNT);
+
+        IncomingWireAchMapping matchingIncomingWireAchMapping = null;
+        Collection<IncomingWireAchMapping> incomingWireAchMappings = businessObjectService.findAll(IncomingWireAchMapping.class);
+
+        for (IncomingWireAchMapping mapping : incomingWireAchMappings) {
+            List<AchIncomeNote> notes = transaction.getNotes();
+            if (doNotesMatch(mapping, notes)) {
+                matchingIncomingWireAchMapping = mapping;
+                break;
+            }
+        }
+
+        if (ObjectUtils.isNotNull(matchingIncomingWireAchMapping)) {
+            chart = matchingIncomingWireAchMapping.getChartOfAccountsCode();
+            objectCode = matchingIncomingWireAchMapping.getFinancialObjectCode();
+            account = matchingIncomingWireAchMapping.getAccountNumber();
+        }
+
+        setupSourceAccountingLine(transaction, advanceDepositDocument, chart, objectCode, account);
+    }
+
+    protected boolean doNotesMatch(IncomingWireAchMapping mapping, List<AchIncomeNote> notes) {
+        final String regex = ".*" + mapping.getShortDescription() + ".*";
```

Comment:
Good suggestions, thanks. Let me try the String indexOf() search first since that would be simpler than the regex, and if not, I'll see about escaping any special characters before the search.